File: Java/ToDoList.java
package Java;

import java.util.Scanner;

public class ToDoList {

    public static void main(String[] args) {

            Scanner scan = new Scanner(System.in);

            String[][][] toDo = new String[12][31][8];

            boolean exit = false;
            byte option;
            while (!exit) {
                System.out.println("Enter 1: add a new appointment!");
                System.out.println("Enter 2: verify appointment!");
                System.out.println("Enter 0: exit!");

                option = scan.nextByte();

                if (option == 1) { // add toDo
                    boolean monthValid = false;
                    int month = 0;
                    while (!monthValid) {
                        System.out.println("Enter to month: ");
                        month = scan.nextInt();
                        if (month > 0 && month <= 12) {
                            monthValid = true;
                        } else {
                            System.out.println("Month invalid, type it again!");
                        }
                    }

                    boolean dayValid = false;
                    int day = 0;
                    while (!dayValid) {
                        System.out.println("Enter to day: ");
                        day = scan.nextInt();
                        if (day > 0 && day <= 31) {
                            dayValid = true;
                        } else {
                            System.out.println("Day invalid, type it again!");
                        }
                    }

                    boolean hourValid = false;
                    int hour = 0;
                    while (!hourValid) {
                        System.out.println("Enter toDo hour: ");
                        hour = scan.nextInt();
                        if (hour >= 0 && hour <= 8) {
                            hourValid = true;
                        } else {
                            System.out.println("Hour invalid, type again!");
                        }
                    }
                    month--;
                    day--; // decrement 0 and start couting day 1 on the exact line
                    System.out.println("Enter toDo: ");
                    toDo[month][day][hour] = scan.next();

                } else if (option == 2) { // check toDo

                    boolean monthValid = false;
                    int month = 0;
                    while (!monthValid) {
                        System.out.println("Enter with month: ");
                        month = scan.nextInt();
                        if (month > 0 && month <= 12) {
                            monthValid = true;
                        } else {
                            System.out.println("Month invalid, type it again!");
                        }
                    }

                    boolean dayValid = false;
                    int day = 0;
                    while (!dayValid) {
                        System.out.println("Enter with day: ");
                        day = scan.nextInt();
                        if (day > 0 && day <= 31) {
                            dayValid = true;
                        } else {
                            System.out.println("Day invalid, type it again!");
                        }
                    }

                    boolean hourValid = false;
                    int hour = 0;
                    while (!hourValid) {
                        System.out.println("Enter hour toDo: ");
                        hour = scan.nextInt();
                        if (hour >= 0 && hour <= 24) {
                            hourValid = true;
                        } else {
                            System.out.println("Hour invalid, type it again!");
                        }
                    }

                    month--;
                    day--;
                    System.out.println("The toDo schedule is: ");
                    System.out.println(toDo[month][day][hour]);
                } else if (option == 0) {
                    exit = true;

                } else {
                    System.out.println("Option invalid, type it again.");
                }
            }
        }
    }
}

File: Python/ToDoList.py
to_do = [[[None for _ in range(8)] for _ in range(31)] for _ in range(12)]

exit_program = False

while not exit_program:
    print("Enter 1: add a new appointment!")
    print("Enter 2: verify appointment!")
    print("Enter 0: exit!")

    option = int(input())

    if option == 1:  # Add toDo
        month_valid = False
        month = 0
        while not month_valid:
            print("Enter the month: ")
            month = int(input())
            if 1 <= month <= 12:
                month_valid = True
            else:
                print("Month invalid, type it again!")

        day_valid = False
        day = 0
        while not day_valid:
            print("Enter the day: ")
            day = int(input())
            if 1 <= day <= 31:
                day_valid = True
            else:
                print("Day invalid, type it again!")

        hour_valid = False
        hour = 0
        while not hour_valid:
            print("Enter toDo hour: ")
            hour = int(input())
            if 0 <= hour <= 7:
                hour_valid = True
            else:
                print("Hour invalid, type it again!")

        month -= 1
        day -= 1  # Decrement 1 to start counting from 0
        print("Enter toDo: ")
        to_do[month][day][hour] = input()

    elif option == 2:  # Check toDo
        month_valid = False
        month = 0
        while not month_valid:
            print("Enter the month: ")
            month = int(input())
            if 1 <= month <= 12:
                month_valid = True
            else:
                print("Month invalid, type it again!")

        day_valid = False
        day = 0
        while not day_valid:
            print("Enter the day: ")
            day = int(input())
            if 1 <= day <= 31:
                day_valid = True
            else:
                print("Day invalid, type it again!")

        hour_valid = False
        hour = 0
        while not hour_valid:
            print("Enter hour toDo: ")
            hour = int(input())
            if 0 <= hour <= 7:
                hour_valid = True
            else:
                print("Hour invalid, type it again!")

        month -= 1
        day -= 1
        print("The toDo schedule is: ")
        print(to_do[month][day][hour])

    elif option == 0:
        exit_program = True

    else:
        print("Option invalid, type it again.")
